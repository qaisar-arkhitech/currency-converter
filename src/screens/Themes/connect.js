import {connect} from "react-redux"

import {
  selectDefaultTheme,
  selectStyleableTheme,
  selectThemes,
} from "../../redux/reducers/themes/selectors"
import {createChangeThemeAction} from "../../redux/reducers/themes/actionCreators"

const mapStateToProps = (state) => ({
  themes: selectThemes(state),
  defaultTheme: selectDefaultTheme(state),
  styleableTheme: selectStyleableTheme(state),
})

const mapDispatchToProps = {
  changeTheme: createChangeThemeAction,
}

export default connect(mapStateToProps, mapDispatchToProps)
