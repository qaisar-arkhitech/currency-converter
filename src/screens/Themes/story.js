import React from "react"
import Themes from "."

export default {
  title: "Themes",
  storyFn() {
    return (
      <div>
        <Themes />
      </div>
    )
  },
}
