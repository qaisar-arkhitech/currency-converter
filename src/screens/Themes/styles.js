import {StyleSheet} from "react-native"

// Themes: Styling here
const styles = (styleableTheme) => {
  return StyleSheet.create({
    root: {
      flex: 1,
      backgroundColor: styleableTheme[50],
    },
  })
}

export default styles
