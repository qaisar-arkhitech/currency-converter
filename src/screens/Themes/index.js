import React from "react"
import {View, SafeAreaView, ScrollView, StatusBar} from "react-native"
import PropTypes from "prop-types"
import flow from "lodash/flow"
import Entypo from "react-native-vector-icons/Entypo"
import Color from "color"
import themeStyles from "./styles"
import connect from "./connect"
import {RowItem, RowSeparator} from "../../components/RowItem"

// Themes Component content
export const Themes = ({themes, changeTheme, defaultTheme, styleableTheme}) => {
  const colors = Object.keys(themes) || []
  const styles = themeStyles(styleableTheme)
  return (
    <SafeAreaView style={styles.root}>
      <StatusBar barStyle="default" backgroundColor={styleableTheme[500]} />
      <ScrollView>
        {colors.map((color, index) => (
          <React.Fragment key={`theme-${index + 1}`}>
            <RowItem
              title={`Theme ${index + 1}`}
              onPress={() => changeTheme(color)}
              rightIcon={
                <View
                  style={{
                    flexDirection: "row",
                  }}>
                  {defaultTheme === color ? (
                    <View style={{marginRight: 12}}>
                      <Entypo name="check" size={20} />
                    </View>
                  ) : null}
                  <View
                    style={{
                      width: 20,
                      height: 20,
                      backgroundColor: color,
                    }}
                  />
                </View>
              }
            />
            <RowSeparator
              style={{
                backgroundColor: Color(styleableTheme[500]).darken(0.04).hex(),
              }}
            />
          </React.Fragment>
        ))}
      </ScrollView>
    </SafeAreaView>
  )
}

// Themes Proptypes
Themes.propTypes = {
  themes: PropTypes.object,
  changeTheme: PropTypes.func,
  defaultTheme: PropTypes.string,
  styleableTheme: PropTypes.object,
}

// Themes Default props
Themes.defaultProps = {
  themes: {},
  changeTheme: () => {},
  defaultTheme: "",
  styleableTheme: {},
}

export default flow([connect])(Themes)
