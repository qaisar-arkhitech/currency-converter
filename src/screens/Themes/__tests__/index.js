import React from "react"
import renderer from "react-test-renderer"
import Themes from ".."

const createProps = () => ({
  classes: {},
})

describe("<Themes />", () => {
  it("Does not explode", () => {
    const props = createProps()
    const component = renderer.create(<Themes {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
