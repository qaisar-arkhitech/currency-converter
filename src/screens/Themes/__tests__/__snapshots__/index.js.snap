// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Themes /> Does not explode 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#e6f7ee",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          Theme 1
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 12,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Entypo",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ï…€
            </Text>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#2fbe74",
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 0.5,
            },
            Object {
              "backgroundColor": "#D8F2E4",
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          Theme 2
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#8b50da",
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 0.5,
            },
            Object {
              "backgroundColor": "#D8F2E4",
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          Theme 3
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#f69400",
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 0.5,
            },
            Object {
              "backgroundColor": "#D8F2E4",
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          Theme 4
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#000000",
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 0.5,
            },
            Object {
              "backgroundColor": "#D8F2E4",
            },
          ]
        }
      />
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
